extensions/src/main/java/org/pentaho/platform/web/http/api/resources/RepositoryResource.java

```diff
@@ -650,6 +652,17 @@ private boolean hasOperationId( final List<IPluginOperation> operations, final S
 
   protected Response doService( String contextId, String resourceId ) throws ObjectFactoryException,
       PluginBeanException, IOException, URISyntaxException {
+    // Check if we are editing a content
+    if(resourceId != null && (resourceId.equals("editor") || resourceId.equals("edit"))) {
+       // Check if user has permission to edit the content
+       if(!canEdit()) {
+         logger.error( MessageFormat.format( "User is not allowed to edit the content", contextId ) );
+         return Response.status(Status.UNAUTHORIZED).build();
+       }
+    }
```

Comment:
One other related issue that I realized is that, as is, the "no permissions" error would be returned _before_ the _not found_ one, which is returned at the last line of this method. We could be returning a no permissions error for a resource which does not really exist!

Reply:
If put in this place, it would effectively only apply to Content Generators and would be done after knowing that the content generator really exists: https://github.com/pentaho/pentaho-platform/blob/9.3/extensions/src/main/java/org/pentaho/platform/web/http/api/resources/RepositoryResource.java#L892

Reply:
In that location, the `resourceId` should be stored in `fac.getCommand()`.

Reply:
good suggestion. I will implement this change